classes/issue/Collector.inc.php

```diff
@@ -0,0 +1,460 @@
+    {
+        $this->isPublished = $isPublished;
+        return $this;
+    }
+
+    /**
+     * Set volumes filter
+     *
+     * @param int[] $volumes
+     *
+     * @return $this
+     */
+    public function filterByVolumes(array $volumes): self
+    {
+        if (!is_null($volumes) && !is_array($volumes)) {
+            $volumes = [$volumes];
+        }
+        $this->volumes = $volumes;
+        return $this;
+    }
+
+    /**
+     * Set volumes filter
+     *
+     * @param int[] $numbers
+     *
+     * @return $this
+     */
+    public function filterByNumbers(array $numbers): self
+    {
+        if (!is_null($numbers) && !is_array($numbers)) {
+            $numbers = [$numbers];
+        }
+        $this->numbers = $numbers;
+        return $this;
+    }
+
+    /**
+     * Set volumes filter
+     *
+     * @param int[] $years
+     *
+     * @return $this
+     */
+    public function filterByYears(array $years): self
+    {
+        if (!is_null($years) && !is_array($years)) {
+            $years = [$years];
+        }
+        $this->years = $years;
+        return $this;
+    }
+
+    /**
+     * Set issue id filter
+     *
+     * @return $this
+     */
+    public function filterByIds(array $issueIds): self
+    {
+        $this->issueIds = $issueIds;
+        return $this;
+    }
+
+    /**
+     * set urlPath filter
+     *
+     * @return $this
+     */
+    public function filterByUrlPath(string $urlPath): self
+    {
+        $this->urlPath = $urlPath;
+        return $this;
+    }
+
+    /**
+     * Set titles filter
+     *
+     * @return $this
+     */
+    public function filterByTitles(array $titles): self
+    {
+        $this->titles = $titles;
+        return $this;
+    }
+
+    /**
+     * Set query search phrase
+     *
+     * @return $this
+     */
+    public function searchPhrase(string $phrase): self
+    {
+        $this->searchPhrase = $phrase;
+        return $this;
+    }
+
+    /**
+     * Limit the number of objects retrieved
+     *
+     * @return $this
+     */
+    public function limit(int $count): self
+    {
+        $this->count = $count;
+        return $this;
+    }
+
+    /**
+     * Offset the number of objects retrieved, for example to
+     * retrieve the second page of contents
+     *
+     * @return $this
+     */
+    public function offset(int $offset): self
+    {
+        $this->offset = $offset;
+        return $this;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getQueryBuilder(): Builder
+    {
+        $this->columns[] = 'i.*';
+        $q = DB::table($this->dao->table, 'i')
+            ->leftJoin('issue_settings as iss', 'i.issue_id', '=', 'iss.issue_id')
+            ->leftJoin('custom_issue_orders as o', 'o.issue_id', '=', 'i.issue_id')
+            // TODO: Is this still necessary (not just for search phrase, but also others)
+            ->groupBy('i.issue_id', $this->orderColumn);
```

Comment:
The way it's currently coded, yes; if there are multiple `issue_settings` rows for a single `issues` entry (which is likely), the `LEFT JOIN` will result in several rows for each issue, thus the `GROUP BY` is being used to collapse the result set into a single row per issue.

To resolve this, you'll need to make sure you are only making joins that do not result in multiple rows matching. For example, there will never be more than one more row per issue for...
```
SELECT i.* FROM issues i JOIN journals j ON (i.journal_id = j.journal_id);
```
However, this will potentially return several rows per issue:
```
SELECT i.* FROM issues i JOIN issue_settings iss ON (i.issue_id = iss.issue_id);
```
It looks like the `iss` join has already been converted below to a `WHERE i.issue_id IN (some subquery)`, so I think the `iss` join above is actually not used.